### PR TITLE
Check permissions for command list + completions

### PR DESCRIPTION
**Pull request by Kumpelinus**

### CODE REVIEW

**Comment by Kumpelinus** [comment]
Features work already. Although I might want to change a few function signatures, etc still.

**Comment by Kumpelinus** [comment]
When done this should fix #617.

**Comment by andrewgazelka** [comment]
this is a draft PR want me to look at it @Kumpelinus

File: crates/hyperion-command/src/component.rs [hunk 1]
```diff
@@ -25,6 +26,19 @@ impl CommandRegistry {
     pub fn all(&self) -> impl Iterator<Item = &str> {
         self.commands.keys().map(String::as_str)
     }
+
+    pub fn get_permitted(&self, world: &World, caller: Entity) -> impl Iterator<Item = &str> {
```
Comment (andrewgazelka) [inline crates/hyperion-command/src/component.rs:30]: maybe add some documentation on what the `&str` is (it is commands)

**Review by andrewgazelka — APPROVED** [review]
LGTM

**Comment by Kumpelinus** [comment]
I marked it as a draft originally because I thought I would make quite a few more changes, but I think it might already be fine as it is.

**Pull request merged**